aten/src/ATen/native/mps/operations/ScatterGather.mm

```diff
@@ -189,57 +162,31 @@
   @autoreleasepool {
 
     MPSShape* input_shape = getMPSShape(self);
-    NSString* ns_input_shape_key = [[input_shape valueForKey:@"description"] componentsJoinedByString:@","];
     MPSShape* index_shape = getMPSShape(index);
-    NSString* ns_index_shape_key = [[index_shape valueForKey:@"description"] componentsJoinedByString:@","];
     MPSShape* src_shape = getMPSShape(src);
-    NSString* ns_src_shape_key = [[src_shape valueForKey:@"description"] componentsJoinedByString:@","];
-
-    int num_input_dims = [input_shape count];
-    int num_index_dims = [index_shape count];
-    int num_src_dims = [src_shape count];
+    uint32_t num_input_dims = [input_shape count];
+    uint32_t num_index_dims = [index_shape count];
+    uint32_t num_src_dims = [src_shape count];
 
     TORCH_CHECK(num_input_dims == num_index_dims && num_index_dims == num_src_dims, "Input, index and src must have same rank")
 
     // Do we need to slice into the src tensor?
     bool needSlice = false;
     bool inputNeedSlice = false;
 
-    for(int i = 0; i < num_input_dims; i++) {
+    for(uint32_t i = 0; i < num_input_dims; i++) {
```

Comment:
nitpick :)  
the pytorch coding style for these for loops would be `for (const auto i : c10::irange(num_input_dims)) {...}`